orthoinference/src/main/java/org/reactome/orthoinference/StableIdentifierGenerator.java

```diff
@@ -0,0 +1,67 @@
+package org.reactome.orthoinference;
+
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.HashMap;
+import java.util.Map;
+import static org.gk.model.ReactomeJavaConstants.*;
+
+/*
+ *  All PhysicalEntitys, ReactionlikeEvents and Pathways are routed to this class to generate their stable identifiers
+ */
+public class StableIdentifierGenerator {
+
+    private static MySQLAdaptor dba;
+    private static GKInstance instanceEditInst;
+    private static Map<String,Integer> seenOrthoIds = new HashMap<>();
+    private static String speciesAbbreviation = null;
+
+    public static GKInstance generateOrthologousStableId(GKInstance inferredInst, GKInstance originalInst) throws Exception {
+
+        // All Human PhysicalEntitys and Events will have a StableIdentifier instance in the stableIdentifier attribute
+        GKInstance stableIdentifierInst = (GKInstance) originalInst.getAttributeValue(stableIdentifier);
+
+        // For now, Human is hard-coded as the source species, so we replace the stableIdentifier source species based on that assumption
+        String sourceIdentifier = (String) stableIdentifierInst.getAttributeValue(identifier);
+        String targetIdentifier = sourceIdentifier.replace("HSA", speciesAbbreviation);
```

Comment:
I think the source species was configurable in the old implementation.  Could the same be done here and the sourceSpecies abbreviation taken from the species info file the way it is for the target/inferred species?  

Plant Reactome might want to use the new orthoinference Java implementation, so being able to configure would be helpful for them.

Reply:
Yes thats something I still need to do with the whole of Orthoinference. I kept it with the source species hard-coded actually to keep it in line with the current implementation. In the near future I'm going to do some work to make it agnostic to source species

Reply:
Cool, I can open a GitHub issue so it's on the to do list if you like?

Reply:
That sounds good to me, thanks

Reply:
Done! https://github.com/reactome/data-release-pipeline/issues/30